File: RepoTransBench/target_projects/Python/C++/os_slacker/public_tests/test_public_utilities.cpp
#include <gtest/gtest.h>
#include <string>
#include <vector>
#include <map>

// Dummy implementation
std::string get_item_id_by_name_public(const std::vector<std::map<std::string, std::string>>& list_dict, const std::string& name) {
    for (const auto& item : list_dict) {
        auto itName = item.find("name");
        auto itId = item.find("id");
        if (itName != item.end() && itId != item.end()) {
            if (itName->second == name) return itId->second;
        }
    }
    return "";
}

TEST(TestGetItemIDByNamePublic, test_get_item_id_by_name) {
    std::vector<std::map<std::string, std::string>> list_dict{
        {{"name", "public_channel"}, {"id", "789"}},
        {{"name", "other"}, {"id", "456"}}
    };
    EXPECT_EQ("789", get_item_id_by_name_public(list_dict, "public_channel"));
}

File: RepoTransBench/target_projects/Python/Rust/os_slacker/public_tests/test_public_utilities.rs
use os_slacker::utilities::get_item_id_by_name;

#[test]
fn test_get_item_id_by_name() {
    let list_dict = vec![
        serde_json::json!({"name": "public_channel", "id": "789"}).as_object().unwrap().clone(),
        serde_json::json!({"name": "other", "id": "456"}).as_object().unwrap().clone(),
    ];
    let result = get_item_id_by_name(&list_dict, "public_channel");
    assert_eq!(result.as_deref(), Some("789"));
}